src/Xhgui/Profiles.php

```diff
@@ -22,6 +22,20 @@ public function __construct(MongoDb $db)
      */
     public function insert($profile)
     {
+	    $profile = $this->setProfilingId($profile);
         return $this->_collection->insert($profile, array('w' => 0));
     }
+
+	/**
+	 * Set profiling ID, if it is already initiated
+	 *
+	 * @param array $profile The profile data
+	 * @return array
+	 */
+	public function setProfilingId($profile) {
+		if (!empty(Xhgui_Saver::getLastProfilingId())) {
+			$profile['_id'] = Xhgui_Saver::getLastProfilingId();
+		}
+		return $profile;
+	}
```

Comment:
I think you could just turn this into a static `getCurrentProfileId` method that provides static variable (initialized to `new MongoId` on first call)
and then just call that from `insert` and you'll have the ability to call it from anywhere that requires it.

Does this make sense to you?